### PR TITLE
fix!: inconsistency in CRD for logging configuration fixed

### LINKED ISSUE #736: Vector aggregator discovery config path not consistent with other Stackable operators
### Affected version

23.7.0

### Current and expected behavior

The ZooKeeper operator uses the property clusterConfig.logging.vectorAggregatorConfigMapName for Vector aggregator discovery whereas the other operators use clusterConfig.vectorAggregatorConfigMapName. In ZooKeeper it is configured like this:

```
  clusterConfig:
    logging:
      vectorAggregatorConfigMapName: vector-aggregator-discovery
```

For consistency with other operators it should be expressed like this:

```
  clusterConfig:
    vectorAggregatorConfigMapName: vector-aggregator-discovery
```

### Possible solution

_No response_

### Additional context

_No response_

### Environment

_No response_

### Would you like to work on fixing this bug?

None

### PR DESCRIPTION
**Pull request by fhennig**
# Description

This is a **breaking** CRD change.

fixes #736 

- docs updated

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by maltesander** [comment]
This has to be adapted in the integration tests of products using zookeeper (and maybe demos).

**Comment by fhennig** [comment]
test run: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/zookeeper-operator-it-custom/66/

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ All notable changes to this project will be documented in this file.
 - `vector` `0.26.0` -> `0.33.0` ([#709], [#732]).
 - `operator-rs` `0.44.0` -> `0.52.1` ([#711], [#730], [#731]).
 - Let secret-operator handle certificate conversion ([#695]).
+- removed the `logging` layer in the `clusterConfig` CRD to make logging config consistent with other operators ([#739]).
```
Comment (maltesander) [inline CHANGELOG.md:19]:
```suggestion
- [BREAKING]: removed the `logging` layer in the `clusterConfig` CRD to make logging config consistent with other operators ([#739]).
```

**Review by maltesander — CHANGES_REQUESTED** [review]
There are missing some changes in other integration tests. Just grep like:
```
    logging:
      vectorAggregatorConfigMapName:
```

**Comment by fhennig** [comment]
new test run: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/zookeeper-operator-it-custom/68/

**Review by maltesander — APPROVED** [review]
LGTM!

**Pull request merged**